### ISSUE TITLE
Rewrite throws argument exception when navigating to bare url that ends in multiple slashes

### ISSUE THREAD

**Issue by is12b** [umbraco/Umbraco-CMS#16585]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.3.2, 13.0.3, 10.8.6

### Bug summary

If the bare url ends in more than 1 slash (E.g https://localhost:43100//) the `Umbraco.Extensions.UriExtensions.Rewrite` throws the `ArgumentException("Path must start with a slash.", "path");` and subsequently results in a 500 error.

https://github.com/umbraco/Umbraco-CMS/blob/release-13.3.2/src/Umbraco.Core/Extensions/UriExtensions.cs#L25

### Specifics

![image](https://github.com/umbraco/Umbraco-CMS/assets/8949513/35dce4d1-a1bd-416e-960a-41717020c32a)

### Steps to reproduce

Navigate to any bare url with multiple slashes (/) at the end on a Umbraco installation.

### Expected result / actual result

We would expect it to just remove the additional slashes:
https://localhost:43100//
https://localhost:43100///////////
https://boxit.dk//

---
_This item has been added to our backlog AB#42381_

**Comment by NguyenThuyLan** [comment]
I can reproduce it on both v13 and v14.
1. Create one document type with template and then create root content from it
2. Then go to root link with 2 slash
<img width="1106" alt="image" src="https://github.com/user-attachments/assets/eec1246f-4c87-462f-856b-839d090b8ccd">

**Comment by kjac** [comment]
We should look at this 😄 I do believe we should always resolve unroutable URLs as a 404 Not Found response. We'll add this to an upcoming sprint 👍

**Comment by PerplexDaniel** [comment]
We also ran in this issue for a client of ours where Amazon CloudFront appends an additional `/` to requests which results in visitors ending up at `https://homepage.com//` which then crashes with the same exception mentioned by @is12b.

The root cause seems [this code](https://github.com/umbraco/Umbraco-CMS/blob/b839135e7102770d1b6499d5a04d1116890bcf00/src/Umbraco.Core/Routing/UriUtility.cs#L113):

```csharp
if (path != "/")
{
    path = path.TrimEnd(Constants.CharArrays.ForwardSlash);
}
```

When the path is `//` we enter the `if` statement, which then proceeds to remove *all* trailing slashes.
So `"//`" is turned into `""`, which then results in the Exception thrown by [`UriExtensions.Rewrite`](https://github.com/umbraco/Umbraco-CMS/blob/b839135e7102770d1b6499d5a04d1116890bcf00/src/Umbraco.Core/Extensions/UriExtensions.cs#L47).

I'd say the code can be updated to this to fix the issue. There are other ways to do this like trimming any additional trailing slashes first, but this seems the most simple and straightforward. So when `path` ends up as `''` after trimming trailing `/` we update it to `/` as path always needs to start with a slash.

```csharp
if (path != "/")
{
    path = path.TrimEnd(Constants.CharArrays.ForwardSlash);

    if (path == "")
    {
        // Path must start with a slash.
        path = "/";
    }
}
```

It does appear the intention is to strip any trailing / for other paths, so for example `/about-us/` should always be turned into `/about-us`. The homepage is the exception (hence your `!= "/"` check), but cases like `//`, `///` etc slip through the cracks here and get turned into `""` by accident.

**Comment by kjac** [comment]
Fix in #17596 17596
